src/test/java/com/mongodb/hibernate/internal/translate/mongoast/AstTestUtils.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.translate.mongoast;
+
+import com.mongodb.hibernate.internal.translate.mongoast.filter.AstComparisonFilterOperation;
+import com.mongodb.hibernate.internal.translate.mongoast.filter.AstComparisonFilterOperator;
+import com.mongodb.hibernate.internal.translate.mongoast.filter.AstFieldOperationFilter;
+import com.mongodb.hibernate.internal.translate.mongoast.filter.AstFilterFieldPath;
+import org.bson.BsonValue;
+
+public final class AstTestUtils {
```

Comment:
The fact that `AstTestUtils` was introduced in this PR, and yet, wasn't used in `AstNorFilterTests`, also introduced in this PR, shows how such adhoc convenience APIs for tests easily get unnoticed, unknown and unused by the maintainers. Thus, let's not try and predict that the method `AstTestUtils.createFieldOperationFilter` will be used in many different `mongoast` tests by putting this class in the `com.mongodb.hibernate.internal.translate.mongoast` package. Instead,

- let's shove it as deep as we can, which is in `com.mongodb.hibernate.internal.translate.mongoast.filter`;
- remove `public` from the class and its method;
- rename it to something like `FilterTestUtils`.

Reply:
I was going to comment opposing this, because I thought `assertSelectionQuery` would end up extracted into the Utils class (I expect it should be used in the ensuing sort etc. PRs). However, I see this has only filter in it, so if that is the only method, I agree.

Anyway, let's just make sure we use the `assert...` pattern from this PR in future PRs, using shared code. It is fine and probably makes more sense to have that code shared in the next PR.

Reply:
> I thought assertSelectionQuery would end up extracted into the Utils class

We already have `com.mongodb.hibernate.MongoTestAssertions` (named after `com.mongodb.hibernate.internal.MongoTestAssertions`) for test-specific custom assertions used by different test classes.

Reply:
changed as Valentin suggested.

Reply:
The "remove `public` from the class and its **method**" part is missing. The method `createFieldOperationFilter` does not need to be `public`, let's change that.

Reply:
done